src/utilities/effects.ts

```diff
@@ -4682,15 +4688,35 @@ const effects = {
       if (query !== null) {
         const viewIdAsNumber = getSearchParameterNumber(SearchParameters.VIEW_ID, query);
 
+        // Derive view from url or model default
+        let view;
         if (viewIdAsNumber !== null) {
           const data = await reqHasura<View>(gql.GET_VIEW, { id: viewIdAsNumber }, user);
-          const { view } = data;
+          const { view: fetchedView } = data;
+          view = fetchedView;
+        } else if (defaultView !== null && defaultView !== undefined) {
+          view = defaultView;
+        }
 
-          if (view !== null) {
+        if (view) {
+          // Return view if not asked to migrate the view
+          if (!migrate) {
             return view;
           }
-        } else if (defaultView !== null && defaultView !== undefined) {
-          return defaultView;
+
+          // Otherwise perform any needed migrations
+          const { migratedView, error, anyMigrationsApplied } = await applyViewMigrations(view);
+          if (migratedView && anyMigrationsApplied) {
+            await effects.updateView(migratedView.id, { definition: migratedView.definition }, user);
```

Comment:
Should this actually update the view in the DB? I could see it being worrying to the user when they open an old view and suddenly gets a "View Updated Successfully" toast. Then they waste time trying to figure out what they accidentally changed.

Reply:
That's what we had settled on right @dandelany @mattdailis @Mythicaeda? We could also potentially reduce confusion by changing or hiding the message?

Reply:
👍 yep updating the view in the DB is correct. I'd be OK with changing the message to "View Migrated Successfully" or hiding it for this case, but not critical.

Reply:
Maybe something like "View Automatically Migrated Successfully" to help indicate that the user did not specifically do an action to cause the toast?

Reply:
Yeah I like that.

Reply:
Is there an error message? If not then `View Automatically Migrated` is sufficient IMO. Otherwise, yeah, as Dan and Aaron said, this should update the DB.

Reply:
There's an error message when the view can't be successfully migrated (really when the migration script fails) but if the view is otherwise invalid you'll just get failures either in SSR or later on in the app since we don't validate existing views against the schema before load. Could revisit this later but would probably break even more views out there.